book/modules.md

```diff
@@ -147,73 +145,63 @@ hi there!
 ## Environment Variables
 
 So far we used modules just to import custom commands.
-It is possible to export environment variables the same way.
-The syntax is slightly different than what you might be used to from commands like [`let-env`](commands/let-env.md) or [`load-env`](commands/load-env.md):
+However, modules can also define an environment using [`export-env`](commands/export-env.md):
 
 ```
 # greetings.nu
 
-export env MYNAME { "Arthur, King of the Britons" }
+export-env {
+    let-env MYNAME = "Arthur, King of the Britons"
+}
 
-export def hello [name: string] {
-    $"hello ($name)"
+export def hello [] {
+    $"hello ($env.MYNAME)"
 }
 ```
 
-`use` works the same way as with custom commands:
-
+`use` will run the code inside the `export-env` block and merge its environment into the current scope:
 ```
 > use greetings.nu
 
-> $env."greetings MYNAME"
+> $env.MYNAME
 Arthur, King of the Britons
 
-> greetings hello $env."greetings MYNAME"
+> greetings hello
 hello Arthur, King of the Britons!
 ```
 
-You can notice we do not assign the value to `MYNAME` directly.
-Instead, we give it a block of code (`{ ...}`) that gets evaluated every time we call [`use`](commands/use.md).
-We can demonstrate this property, for example, with the [`random`](commands/random.md) command:
-
+::: tip
+You might wonder why we can't just define `let-env` at the top of the module.
+The reason is that the `export-env {...}` block keeps its scope separate from the rest of the module which makes it more organized.
+You can put a complex code defining your environment without polluting the namespace of the module, for example:
 ```
-> module roll { export env ROLL { random dice | into string } }
-
-> use roll ROLL
-
-> $env.ROLL
-4
-
-> $env.ROLL
-4
-
-> use roll ROLL
+export-env {
+    def tmp [] { "tmp" }
 
-> $env.ROLL
-6
+    let len = (tmp | str length)
 
-> $env.ROLL
-6
+    load-env {
+        TMP_LEN: $len
+    }
```

Comment:
This helps! However, as a user reading this I'm now wondering why I have to use `load-env`? Why don't I just write

```
let-env TMP_LEN = $len
```

here, seeing as above we wrote

```
export-env {
    let-env MYNAME = "Arthur, King of the Britons"
}
```

?

Reply:
It's just an example that illustrates you can bring in the environment in any way, not just `let-env`.

Reply:
OK I see. I think this section would be even clearer if it was something like:
```
export-env {
    def name [] { "Arthur" }
    def title [] { "King of the Britons" }

    let-env MYNAME = $'(name), (title)'
}
```

Otherwise it risks seeming like `load-env` must be used when the `export-env` block contains computed values. But it's a fairly minor thing, just a suggestion.